### PR TITLE
sentSpawn functionality

### PR DESCRIPTION
**Pull request by deltamolfar**
(Re-PR because I accidentally deleted previous fork)
Added ``:GetBool()``, and review changes.

- Finally a way for E2 to spawn sents.
- For QoL, limited to wire/native gmod sents + entity tab entities, if `wire_expression2_propcore_whitelist <= 0`)
- Thirdparty addons support (ability to add/remove ents to be able to be spawned within whitelist + a hook with same logic as `Expression2_CanEffect`
- Entity registration is straightforward, and the sentSpawn does all the E2 to Lua castings itself (No need to put every check in _preFactory'ies).
- Ability to spawn **ALL** wire entities (if I didn't forget any).
- Few helper functions to sentSpawn.

While most of registration code/logic is from starfall `prop_sent.lua`, I've found and fixed ~10 sents + added ~6 that weren't originally there.

Current minor bugs:
- gmod_wire_dynamic_button doesn't spawn correctly (commented out).
- gmod_cameraprop and gmod_emitter spawn with no issue, but don't work as expected (commented out).
- Undo for some reason doesn't return `E2 sent (classname)`, but `e2_spawned_sent` when user cancels sent.

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/prop.lua [hunk 1]
```diff
@@ -194,8 +194,402 @@ local function boneVerify(self, bone)
...
+				return Color(e2Value[1], e2Value[2], e2Value[3], e2Value[4])
+			elseif e2Value.r and e2Value.g and e2Value.b then
+				if e2Value.a then return Color(e2Value.r, e2Value.g, e2Value.b, e2Value.a) end
+				return Color(e2Value.r, e2Value.g, e2Value.b)
+			end
+		end
+
+		return nil
+	end,
+}
```
Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/custom/prop.lua:401]: This should be e2 core code rather than local to prop lib.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/entities/gmod_wire_expression2/core/custom/prop.lua (hunk 1 above)
Comment (deltamolfar) [reply lua/entities/gmod_wire_expression2/core/custom/prop.lua:401]: Yeah, I wanted to make it E2Lib code, but wasn't sure, and never got reply in Discord. Will move it

**Comment by thegrb93** [comment]
Looks fine, might do a diff against starfall's to sanity check.

**Comment by thegrb93** [comment]
I think sents_default_params.lua needs to be put somewhere outside of lua/entities. putting lua files there registers the filename as an entity class.

**Review by thegrb93 — CHANGES_REQUESTED** [review]
I think sents_default_params.lua needs to be put somewhere outside of lua/entities. putting lua files there registers the filename as an entity class.

**Comment by deltamolfar** [comment]
- Alright, I've moved it and renamed sents_default_params.lua out of entities folder, to lua/wire/server/sents_registry.lua.
- Changed a whitelist/registry logic a bit (reinforced the registry reusability by thirdparty addons, by separating registering wire elements, and blacklisting).
- Few QoL fixes.

File: lua/wire/server/sents_registry.lua [hunk 2]
```diff
@@ -33,43 +38,56 @@
...
+-- It've been made global to support thirdparty addons.
 ---@param class string
-function SentSpawn.WhitelistRemove(class)
-	if not list.HasEntry("wire_spawnable_ents_whitelist", class) then ErrorNoHaltWithStack("Trying to remove entity that is not registered!") return end
+function SentSpawn.Unregister(class)
+	if not list.HasEntry("wire_spawnable_ents_registry", class) then ErrorNoHaltWithStack("Trying to remove entity that is not registered!") return end
 
-	local whitelist = list.GetForEdit("wire_spawnable_ents_whitelist")
+	local whitelist = list.GetForEdit("wire_spawnable_ents_registry")
 	table.remove(whitelist, class)
```
Comment (thegrb93) [inline lua/wire/server/sents_registry.lua:85]: Shouldn't this be `whitelist[class]=nil`? I don't think table.remove works on string indexed tables.

Comment (deltamolfar) [reply lua/wire/server/sents_registry.lua:85]: Yeah you're right, my bad

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**